executor/distsql.go

```diff
@@ -824,14 +827,23 @@ func (w *tableWorker) executeTask(ctx context.Context, task *lookupTableTask) er
 		sort.Sort(task)
 	}
 
-	if w.isCheckOp && handleCnt != len(task.rows) {
-		obtainedHandlesMap := make(map[int64]struct{}, len(task.rows))
-		for _, row := range task.rows {
-			handle := row.GetInt64(w.handleIdx)
-			obtainedHandlesMap[handle] = struct{}{}
+	if handleCnt != len(task.rows) {
+		if w.isCheckOp {
+			obtainedHandlesMap := make(map[int64]struct{}, len(task.rows))
+			for _, row := range task.rows {
+				handle := row.GetInt64(w.handleIdx)
+				obtainedHandlesMap[handle] = struct{}{}
+			}
+			return errors.Errorf("inconsistent index %s handle count %d isn't equal to value count %d, missing handles %v in a batch",
+				w.idxLookup.index.Name.O, handleCnt, len(task.rows), GetLackHandles(task.handles, obtainedHandlesMap))
+		}
+
+		if len(w.idxLookup.tblPlans) == 1 {
+			// table scan in double read can never has conditions according to convertToIndexScan.
```

Comment:
> table scan in double read can never has conditions according to convertToIndexScan.
the comments is incorrect?

Reply:
What's wrong? `conditions` -> `table filters`?

Reply:
```
TiDB(root@127.0.0.1:test) > desc select * from t where a = 1 and b = 1 and c = 1;
+---------------------+-------+------+----------------------------------------------------------------------+
| id                  | count | task | operator info                                                        |
+---------------------+-------+------+----------------------------------------------------------------------+
| IndexLookUp_11      | 0.00  | root |                                                                      |
| ├─IndexScan_8       | 0.10  | cop  | table:t, index:a, b, range:[1 1,1 1], keep order:false, stats:pseudo |
| └─Selection_10      | 0.00  | cop  | eq(test.t.c, 1)                                                      |
|   └─TableScan_9     | 0.10  | cop  | table:t, keep order:false, stats:pseudo                              |
+---------------------+-------+------+----------------------------------------------------------------------+
4 rows in set (0.00 sec)
```

it can have a filter?

Reply:
The `table scan` only represents `TableScan_9` in this plan and `Selection_10` is excluded by `len(w.idxLookup.tblPlans) == 1`.

Reply:
okay